src/components/dashboard/RecoveryInProgress/index.tsx

```diff
@@ -0,0 +1,116 @@
+import { Box, Card, Grid, Typography } from '@mui/material'
+import type { ReactElement } from 'react'
+
+import { useAppSelector } from '@/store'
+import { selectRecovery } from '@/store/recoverySlice'
+import { useBlockTimestamp } from '@/hooks/useBlockTimestamp'
+import { WidgetContainer, WidgetBody } from '../styled'
+import RecoveryPending from '@/public/images/common/recovery-pending.svg'
+import ExternalLink from '@/components/common/ExternalLink'
+import { useHasFeature } from '@/hooks/useChains'
+import { FEATURES } from '@/utils/chains'
+
+export function RecoveryInProgress(): ReactElement | null {
+  const blockTimestamp = useBlockTimestamp()
```

Comment:
I've added a `Skeleton` in 9d5270efb105eb11092056396d79566526751e06.

Reply:
Can we adjust it so the Skeleton is only displayed if the safe has recovery enabled? I can see it flashing now even in safes where no recovery module is enabled.

Reply:
Loading recovery takes longer than the timestamp so we do not know whether it is enabled. I would sooner suggest we opt for no loader, but dispatch a notification as well in case the user navigates away. What do you think @TanyaEfremova?

(Due to capacity, I will remove this for now and adjust it in a followup PR as to remove blockers.)